include/ews/ews.hpp

```diff
@@ -9429,238 +9429,1440 @@ namespace ews
                 else if (compare("FullName", std::strlen("FullName"),
                                  child->local_name(), child->local_name_size()))
                 {
-                    fullname = std::string(child->value(), child->value_size());
+                    fullname = std::string(child->value(), child->value_size());
+                }
+                else if (compare("Nickname", std::strlen("Nickname"),
+                                 child->local_name(), child->local_name_size()))
+                {
+                    nickname = std::string(child->value(), child->value_size());
+                }
+            }
+
+            return complete_name(title, firstname, middlename, lastname, suffix,
+                                 initials, fullname, nickname);
+        }
+
+        const std::string& get_title() const EWS_NOEXCEPT { return title_; }
+        const std::string& get_first_name() const EWS_NOEXCEPT
+        {
+            return firstname_;
+        }
+        const std::string& get_middle_name() const EWS_NOEXCEPT
+        {
+            return middlename_;
+        }
+        const std::string& get_last_name() const EWS_NOEXCEPT
+        {
+            return lastname_;
+        }
+        const std::string& get_suffix() const EWS_NOEXCEPT { return suffix_; }
+        const std::string& get_initials() const EWS_NOEXCEPT
+        {
+            return initials_;
+        }
+        const std::string& get_full_name() const EWS_NOEXCEPT
+        {
+            return fullname_;
+        }
+        const std::string& get_nickname() const EWS_NOEXCEPT
+        {
+            return nickname_;
+        }
+
+    private:
+        std::string title_;
+        std::string firstname_;
+        std::string middlename_;
+        std::string lastname_;
+        std::string suffix_;
+        std::string initials_;
+        std::string fullname_;
+        std::string nickname_;
+    };
+
+    enum class email_address_key
+    {
+        email_address_1,
+        email_address_2,
+        email_address_3
+    };
```

Comment:
Why did you change this from `email_address::key` to `email_address_key`? You changed the API during the review without any need and without any notice. Revert this.

Here and other locations.